src/actionCreators/index.js

```diff
@@ -98,9 +98,12 @@ export const updateUser = (userId, userProfile) => {
     axi
       .put(API.UPDATE_SSO_USER(userId), userProfile)
       .then(res => {
+        // Get the updated the current user info after the API request:
+        dispatch(authorize(SELF_SERVER_APP.ROCKETCHAT.NAME, userId));
         return dispatch(updateUserSuccess());
       })
       .catch(err => {
+        dispatch(authorize(SELF_SERVER_APP.ROCKETCHAT.NAME, userId));
```

Comment:
authorizing user even though there is a failure?

Reply:
should have better naming for it, it's not authorizing user but checking the authorization state